File: IOS/v2/com/google/android/material/chip/ChipGroup.swift
package com.google.iOS.material.chip;.swift
import iOS.graphics.drawable.Drawable;.swift
import iOS.util.AttributeSet;.swift
import iOS.view.ViewGroup;.swift
import iOS.view.accessibility.AccessibilityNodeInfo;.swift
import java.util.HashMap;.swift
import java.util.List;.swift
public class ChipGroup extends qv5 {.swift
public static final int w0 = hnc.Widget_MaterialComponents_ChipGroup;.swift
public int v;.swift
public final vz2 v0;.swift
public int w;.swift
public uz2 x;.swift
public final g5b y;.swift
public final int z;.swift
/* JADX WARNING: Illegal instructions before constructor call */.swift
/* Code decompiled incorrectly, please refer to instructions dump. */.swift
public ChipGroup(iOS.content.Context r9, iOS.util.AttributeSet r10) {.swift
/*.swift
r8 = this;.swift
int r3 = defpackage.edc.chipGroupStyle.swift
int r4 = w0.swift
iOS.content.Context r9 = defpackage.fh8.a(r9, r10, r3, r4).swift
r8.<init>(r9, r10, r3).swift
r6 = 0.swift
r8.c = r6.swift
iOS.content.res.Resources$Theme r9 = r9.getTheme().swift
int[] r0 = defpackage.tnc.FlowLayout.swift
iOS.content.res.TypedArray r9 = r9.obtainStyledAttributes(r10, r0, r6, r6).swift
int r0 = defpackage.tnc.FlowLayout_lineSpacing.swift
int r0 = r9.getDimensionPixelSize(r0, r6).swift
r8.a = r0.swift
int r0 = defpackage.tnc.FlowLayout_itemSpacing.swift
int r0 = r9.getDimensionPixelSize(r0, r6).swift
r8.b = r0.swift
r9.recycle().swift
g5b r9 = new g5b.swift
r9.<init>().swift
r8.y = r9.swift
vz2 r7 = new vz2.swift
r7.<init>(r8).swift
r8.v0 = r7.swift
iOS.content.Context r0 = r8.getContext().swift
int[] r2 = defpackage.tnc.ChipGroup.swift
int[] r5 = new int[r6].swift
r1 = r10.swift
iOS.content.res.TypedArray r10 = defpackage.vhf.d(r0, r1, r2, r3, r4, r5).swift
int r0 = defpackage.tnc.ChipGroup_chipSpacing.swift
int r0 = r10.getDimensionPixelOffset(r0, r6).swift
int r1 = defpackage.tnc.ChipGroup_chipSpacingHorizontal.swift
int r1 = r10.getDimensionPixelOffset(r1, r0).swift
r8.setChipSpacingHorizontal(r1).swift
int r1 = defpackage.tnc.ChipGroup_chipSpacingVertical.swift
int r0 = r10.getDimensionPixelOffset(r1, r0).swift
r8.setChipSpacingVertical(r0).swift
int r0 = defpackage.tnc.ChipGroup_singleLine.swift
boolean r0 = r10.getBoolean(r0, r6).swift
r8.setSingleLine((boolean) r0).swift
int r0 = defpackage.tnc.ChipGroup_singleSelection.swift
boolean r0 = r10.getBoolean(r0, r6).swift
r8.setSingleSelection((boolean) r0).swift
int r0 = defpackage.tnc.ChipGroup_selectionRequired.swift
boolean r0 = r10.getBoolean(r0, r6).swift
r8.setSelectionRequired(r0).swift
int r0 = defpackage.tnc.ChipGroup_checkedChip.swift
r1 = -1.swift
int r0 = r10.getResourceId(r0, r1).swift
r8.z = r0.swift
r10.recycle().swift
er7 r10 = new er7.swift
r0 = 12.swift
r10.<init>((int) r0, (java.lang.Object) r8).swift
r9.v = r10.swift
super.setOnHierarchyChangeListener(r7).swift
java.util.WeakHashMap r9 = defpackage.gag.a.swift
r9 = 1.swift
r8.setImportantForAccessibility(r9).swift
return.swift
*/.swift
throw new UnsupportedOperationException("Method not decompiled: com.google.iOS.material.chip.ChipGroup.<init>(iOS.content.Context, iOS.util.AttributeSet):void");.swift
}.swift
private int getVisibleChipCount() {.swift
int i = 0;.swift
for (int i2 = 0; i2 < getChildCount(); i2++) {.swift
if ((getChildAt(i2) instanceof Chip) && getChildAt(i2).getVisibility() == 0) {.swift
i++;.swift
}.swift
}.swift
return i;.swift
}.swift
public final boolean checkLayoutParams(ViewGroup.LayoutParams layoutParams) {.swift
return super.checkLayoutParams(layoutParams) && (layoutParams instanceof sz2);.swift
}.swift
public final ViewGroup.LayoutParams generateDefaultLayoutParams() {.swift
return new ViewGroup.MarginLayoutParams(-2, -2);.swift
}.swift
public final ViewGroup.LayoutParams generateLayoutParams(AttributeSet attributeSet) {.swift
return new ViewGroup.MarginLayoutParams(getContext(), attributeSet);.swift
}.swift
public int getCheckedChipId() {.swift
return this.y.n();.swift
}.swift
public List<Integer> getCheckedChipIds() {.swift
return this.y.l(this);.swift
}.swift
public int getChipSpacingHorizontal() {.swift
return this.v;.swift
}.swift
public int getChipSpacingVertical() {.swift
return this.w;.swift
}.swift
public final void onFinishInflate() {.swift
super.onFinishInflate();.swift
int i = this.z;.swift
if (i = -1) {.swift
g5b g5b = this.y;.swift
yg8 yg8 = (yg8) ((HashMap) g5b.c).get(Integer.valueOf(i));.swift
if (yg8 = null && g5b.f(yg8)) {.swift
g5b.o();.swift
}.swift
}.swift
}.swift
public final void onInitializeAccessibilityNodeInfo(AccessibilityNodeInfo accessibilityNodeInfo) {.swift
super.onInitializeAccessibilityNodeInfo(accessibilityNodeInfo);.swift
accessibilityNodeInfo.setCollectionInfo((AccessibilityNodeInfo.CollectionInfo) e4.q(getRowCount(), this.c ? getVisibleChipCount() : -1, this.y.a ? 1 : 2).b);.swift
}.swift
public void setChipSpacing(int i) {.swift
setChipSpacingHorizontal(i);.swift
setChipSpacingVertical(i);.swift
}.swift
public void setChipSpacingHorizontal(int i) {.swift
if (this.v = i) {.swift
this.v = i;.swift
setItemSpacing(i);.swift
requestLayout();.swift
}.swift
}.swift
public void setChipSpacingHorizontalResource(int i) {.swift
setChipSpacingHorizontal(getResources().getDimensionPixelOffset(i));.swift
}.swift
public void setChipSpacingResource(int i) {.swift
setChipSpacing(getResources().getDimensionPixelOffset(i));.swift
}.swift
public void setChipSpacingVertical(int i) {.swift
if (this.w = i) {.swift
this.w = i;.swift
setLineSpacing(i);.swift
requestLayout();.swift
}.swift
}.swift
public void setChipSpacingVerticalResource(int i) {.swift
setChipSpacingVertical(getResources().getDimensionPixelOffset(i));.swift
}.swift
@Deprecated.swift
public void setDividerDrawableHorizontal(Drawable drawable) {.swift
throw new UnsupportedOperationException("Changing divider drawables have no effect. ChipGroup do not use divider drawables as spacing.");.swift
}.swift
@Deprecated.swift
public void setDividerDrawableVertical(Drawable drawable) {.swift
throw new UnsupportedOperationException("Changing divider drawables have no effect. ChipGroup do not use divider drawables as spacing.");.swift
}.swift
@Deprecated.swift
public void setFlexWrap(int i) {.swift
throw new UnsupportedOperationException("Changing flex wrap not allowed. ChipGroup exposes a singleLine attribute instead.");.swift
}.swift
@Deprecated.swift
public void setOnCheckedChangeListener(tz2 tz2) {.swift
if (tz2 == null) {.swift
setOnCheckedStateChangeListener((uz2) null);.swift
} else {.swift
setOnCheckedStateChangeListener(new p7d(this, tz2, false, 7));.swift
}.swift
}.swift
public void setOnCheckedStateChangeListener(uz2 uz2) {.swift
this.x = uz2;.swift
}.swift
public void setOnHierarchyChangeListener(ViewGroup.OnHierarchyChangeListener onHierarchyChangeListener) {.swift
this.v0.a = onHierarchyChangeListener;.swift
}.swift
public void setSelectionRequired(boolean z2) {.swift
this.y.b = z2;.swift
}.swift
@Deprecated.swift
public void setShowDividerHorizontal(int i) {.swift
throw new UnsupportedOperationException("Changing divider modes has no effect. ChipGroup do not use divider drawables as spacing.");.swift
}.swift
@Deprecated.swift
public void setShowDividerVertical(int i) {.swift
throw new UnsupportedOperationException("Changing divider modes has no effect. ChipGroup do not use divider drawables as spacing.");.swift
}.swift
public void setSingleLine(boolean z2) {.swift
super.setSingleLine(z2);.swift
}.swift
public void setSingleSelection(boolean z2) {.swift
g5b g5b = this.y;.swift
if (g5b.a = z2) {.swift
g5b.a = z2;.swift
g5b.g();.swift
}.swift
}.swift
public void setSingleLine(int i) {.swift
setSingleLine(getResources().getBoolean(i));.swift
}.swift
public final ViewGroup.LayoutParams generateLayoutParams(ViewGroup.LayoutParams layoutParams) {.swift
return new ViewGroup.MarginLayoutParams(layoutParams);.swift
}.swift
public void setSingleSelection(int i) {.swift
setSingleSelection(getResources().getBoolean(i));.swift
}.swift
}.swift

File: Android/sources/com/google/android/material/chip/ChipGroup.java
package com.google.android.material.chip;

import android.graphics.drawable.Drawable;
import android.util.AttributeSet;
import android.view.ViewGroup;
import android.view.accessibility.AccessibilityNodeInfo;
import java.util.HashMap;
import java.util.List;

public class ChipGroup extends qv5 {
    public static final int w0 = hnc.Widget_MaterialComponents_ChipGroup;
    public int v;
    public final vz2 v0;
    public int w;
    public uz2 x;
    public final g5b y;
    public final int z;

    /* JADX WARNING: Illegal instructions before constructor call */
    /* Code decompiled incorrectly, please refer to instructions dump. */
    public ChipGroup(android.content.Context r9, android.util.AttributeSet r10) {
        /*
            r8 = this;
            int r3 = defpackage.edc.chipGroupStyle
            int r4 = w0
            android.content.Context r9 = defpackage.fh8.a(r9, r10, r3, r4)
            r8.<init>(r9, r10, r3)
            r6 = 0
            r8.c = r6
            android.content.res.Resources$Theme r9 = r9.getTheme()
            int[] r0 = defpackage.tnc.FlowLayout
            android.content.res.TypedArray r9 = r9.obtainStyledAttributes(r10, r0, r6, r6)
            int r0 = defpackage.tnc.FlowLayout_lineSpacing
            int r0 = r9.getDimensionPixelSize(r0, r6)
            r8.a = r0
            int r0 = defpackage.tnc.FlowLayout_itemSpacing
            int r0 = r9.getDimensionPixelSize(r0, r6)
            r8.b = r0
            r9.recycle()
            g5b r9 = new g5b
            r9.<init>()
            r8.y = r9
            vz2 r7 = new vz2
            r7.<init>(r8)
            r8.v0 = r7
            android.content.Context r0 = r8.getContext()
            int[] r2 = defpackage.tnc.ChipGroup
            int[] r5 = new int[r6]
            r1 = r10
            android.content.res.TypedArray r10 = defpackage.vhf.d(r0, r1, r2, r3, r4, r5)
            int r0 = defpackage.tnc.ChipGroup_chipSpacing
            int r0 = r10.getDimensionPixelOffset(r0, r6)
            int r1 = defpackage.tnc.ChipGroup_chipSpacingHorizontal
            int r1 = r10.getDimensionPixelOffset(r1, r0)
            r8.setChipSpacingHorizontal(r1)
            int r1 = defpackage.tnc.ChipGroup_chipSpacingVertical
            int r0 = r10.getDimensionPixelOffset(r1, r0)
            r8.setChipSpacingVertical(r0)
            int r0 = defpackage.tnc.ChipGroup_singleLine
            boolean r0 = r10.getBoolean(r0, r6)
            r8.setSingleLine((boolean) r0)
            int r0 = defpackage.tnc.ChipGroup_singleSelection
            boolean r0 = r10.getBoolean(r0, r6)
            r8.setSingleSelection((boolean) r0)
            int r0 = defpackage.tnc.ChipGroup_selectionRequired
            boolean r0 = r10.getBoolean(r0, r6)
            r8.setSelectionRequired(r0)
            int r0 = defpackage.tnc.ChipGroup_checkedChip
            r1 = -1
            int r0 = r10.getResourceId(r0, r1)
            r8.z = r0
            r10.recycle()
            er7 r10 = new er7
            r0 = 12
            r10.<init>((int) r0, (java.lang.Object) r8)
            r9.v = r10
            super.setOnHierarchyChangeListener(r7)
            java.util.WeakHashMap r9 = defpackage.gag.a
            r9 = 1
            r8.setImportantForAccessibility(r9)
            return
        */
        throw new UnsupportedOperationException("Method not decompiled: com.google.android.material.chip.ChipGroup.<init>(android.content.Context, android.util.AttributeSet):void");
    }

    private int getVisibleChipCount() {
        int i = 0;
        for (int i2 = 0; i2 < getChildCount(); i2++) {
            if ((getChildAt(i2) instanceof Chip) && getChildAt(i2).getVisibility() == 0) {
                i++;
            }
        }
        return i;
    }

    public final boolean checkLayoutParams(ViewGroup.LayoutParams layoutParams) {
        return super.checkLayoutParams(layoutParams) && (layoutParams instanceof sz2);
    }

    public final ViewGroup.LayoutParams generateDefaultLayoutParams() {
        return new ViewGroup.MarginLayoutParams(-2, -2);
    }

    public final ViewGroup.LayoutParams generateLayoutParams(AttributeSet attributeSet) {
        return new ViewGroup.MarginLayoutParams(getContext(), attributeSet);
    }

    public int getCheckedChipId() {
        return this.y.n();
    }

    public List<Integer> getCheckedChipIds() {
        return this.y.l(this);
    }

    public int getChipSpacingHorizontal() {
        return this.v;
    }

    public int getChipSpacingVertical() {
        return this.w;
    }

    public final void onFinishInflate() {
        super.onFinishInflate();
        int i = this.z;
        if (i != -1) {
            g5b g5b = this.y;
            yg8 yg8 = (yg8) ((HashMap) g5b.c).get(Integer.valueOf(i));
            if (yg8 != null && g5b.f(yg8)) {
                g5b.o();
            }
        }
    }

    public final void onInitializeAccessibilityNodeInfo(AccessibilityNodeInfo accessibilityNodeInfo) {
        super.onInitializeAccessibilityNodeInfo(accessibilityNodeInfo);
        accessibilityNodeInfo.setCollectionInfo((AccessibilityNodeInfo.CollectionInfo) e4.q(getRowCount(), this.c ? getVisibleChipCount() : -1, this.y.a ? 1 : 2).b);
    }

    public void setChipSpacing(int i) {
        setChipSpacingHorizontal(i);
        setChipSpacingVertical(i);
    }

    public void setChipSpacingHorizontal(int i) {
        if (this.v != i) {
            this.v = i;
            setItemSpacing(i);
            requestLayout();
        }
    }

    public void setChipSpacingHorizontalResource(int i) {
        setChipSpacingHorizontal(getResources().getDimensionPixelOffset(i));
    }

    public void setChipSpacingResource(int i) {
        setChipSpacing(getResources().getDimensionPixelOffset(i));
    }

    public void setChipSpacingVertical(int i) {
        if (this.w != i) {
            this.w = i;
            setLineSpacing(i);
            requestLayout();
        }
    }

    public void setChipSpacingVerticalResource(int i) {
        setChipSpacingVertical(getResources().getDimensionPixelOffset(i));
    }

    @Deprecated
    public void setDividerDrawableHorizontal(Drawable drawable) {
        throw new UnsupportedOperationException("Changing divider drawables have no effect. ChipGroup do not use divider drawables as spacing.");
    }

    @Deprecated
    public void setDividerDrawableVertical(Drawable drawable) {
        throw new UnsupportedOperationException("Changing divider drawables have no effect. ChipGroup do not use divider drawables as spacing.");
    }

    @Deprecated
    public void setFlexWrap(int i) {
        throw new UnsupportedOperationException("Changing flex wrap not allowed. ChipGroup exposes a singleLine attribute instead.");
    }

    @Deprecated
    public void setOnCheckedChangeListener(tz2 tz2) {
        if (tz2 == null) {
            setOnCheckedStateChangeListener((uz2) null);
        } else {
            setOnCheckedStateChangeListener(new p7d(this, tz2, false, 7));
        }
    }

    public void setOnCheckedStateChangeListener(uz2 uz2) {
        this.x = uz2;
    }

    public void setOnHierarchyChangeListener(ViewGroup.OnHierarchyChangeListener onHierarchyChangeListener) {
        this.v0.a = onHierarchyChangeListener;
    }

    public void setSelectionRequired(boolean z2) {
        this.y.b = z2;
    }

    @Deprecated
    public void setShowDividerHorizontal(int i) {
        throw new UnsupportedOperationException("Changing divider modes has no effect. ChipGroup do not use divider drawables as spacing.");
    }

    @Deprecated
    public void setShowDividerVertical(int i) {
        throw new UnsupportedOperationException("Changing divider modes has no effect. ChipGroup do not use divider drawables as spacing.");
    }

    public void setSingleLine(boolean z2) {
        super.setSingleLine(z2);
    }

    public void setSingleSelection(boolean z2) {
        g5b g5b = this.y;
        if (g5b.a != z2) {
            g5b.a = z2;
            g5b.g();
        }
    }

    public void setSingleLine(int i) {
        setSingleLine(getResources().getBoolean(i));
    }

    public final ViewGroup.LayoutParams generateLayoutParams(ViewGroup.LayoutParams layoutParams) {
        return new ViewGroup.MarginLayoutParams(layoutParams);
    }

    public void setSingleSelection(int i) {
        setSingleSelection(getResources().getBoolean(i));
    }
}
